app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -1,17 +1,104 @@
-// package ch.epfl.favo.common;
-//
-// import com.google.firebase.firestore.FirebaseFirestore;
-//
-// public final class DatabaseWrapper {
-//
-//    private static FirebaseFirestore mFirestore;
-//
-//    public DatabaseWrapper() {
-//        FirebaseFirestore.setLoggingEnabled(true);
-//        mFirestore = FirebaseFirestore.getInstance();
-//    }
-//
-//    public static FirebaseFirestore getDatabaseInstance() {
-//        return mFirestore;
-//    }
-// }
+package ch.epfl.favo.common;
+
+import android.util.Log;
+
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreSettings;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class DatabaseWrapper<T extends DatabaseDocument> {
+
+    private static final String TAG = "DatabaseWrapper";
+
+    private static FirebaseFirestore firestore;
+
+    private Class<T> cls;
+
+    private static String collection;
+
+    public DatabaseWrapper(String collection, Class<T> cls) {
+        this.cls = cls;
+        this.collection = collection;
+
+        FirebaseFirestore.setLoggingEnabled(true);
+        FirebaseFirestoreSettings settings =
+                new FirebaseFirestoreSettings.Builder().setPersistenceEnabled(true).build();
+
+        this.firestore = FirebaseFirestore.getInstance();
+        this.firestore.setFirestoreSettings(settings);
```

Comment:
The implementation in my previous comment would avoid applying the settings at a every construction